### PR TITLE
ocamlPackages.uucp: 11.0.0 -> 13.0.0

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

<https://github.com/dbuenzli/uucp/blob/master/CHANGES.md#v1300-2020-03-10-la-forclaz-vs>

Fixed up #112347 and enabled tests.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
Result of `nixpkgs-review pr 113311` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>ocamlPackages.wodan-irmin</li>
  </ul>
</details>
<details>
  <summary>35 packages built:</summary>
  <ul>
    <li>jackline</li>
    <li>ocamlPackages.carton</li>
    <li>ocamlPackages.carton-git</li>
    <li>ocamlPackages.carton-lwt</li>
    <li>ocamlPackages.crowbar</li>
    <li>ocamlPackages.duff</li>
    <li>ocamlPackages.git</li>
    <li>ocamlPackages.git-cohttp</li>
    <li>ocamlPackages.git-cohttp-mirage</li>
    <li>ocamlPackages.git-cohttp-unix</li>
    <li>ocamlPackages.git-unix</li>
    <li>ocamlPackages.index</li>
    <li>ocamlPackages.irmin-containers</li>
    <li>ocamlPackages.irmin-git</li>
    <li>ocamlPackages.irmin-http</li>
    <li>ocamlPackages.irmin-mirage-git</li>
    <li>ocamlPackages.irmin-mirage-graphql</li>
    <li>ocamlPackages.irmin-pack</li>
    <li>ocamlPackages.irmin-unix</li>
    <li>ocamlPackages.jingoo</li>
    <li>ocamlPackages.notty</li>
    <li>ocamlPackages.phylogenetics</li>
    <li>ocamlPackages.printbox</li>
    <li>ocamlPackages.uucp</li>
    <li>ocamlPackages.uuseg</li>
    <li>ocamlformat (ocamlformat_0_16_0)</li>
    <li>ocamlformat_0_11_0</li>
    <li>ocamlformat_0_12</li>
    <li>ocamlformat_0_13_0</li>
    <li>ocamlformat_0_14_0</li>
    <li>ocamlformat_0_14_1</li>
    <li>ocamlformat_0_14_2</li>
    <li>ocamlformat_0_14_3</li>
    <li>ocamlformat_0_15_0</li>
    <li>ocamlformat_0_15_1</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113311` at 76432663 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/123effafa94cb6a30eb3f10e6e4287b8">jackline</a></li></li>
    <li><a href="https://gist.github.com/23709225f802d31cfa4a63830d39a7de">ocamlformat_0_11_0</a></li></li>
    <li><a href="https://gist.github.com/ae55f2a89481ac816a842d1de505140e">ocamlformat_0_12</a></li></li>
    <li><a href="https://gist.github.com/7c2db7b222c385f7b8f69f84762d6ede">ocamlformat_0_13_0</a></li></li>
    <li><a href="https://gist.github.com/0e25e452bd132c9cecf81e835df4d4a8">ocamlformat_0_14_0</a></li></li>
    <li><a href="https://gist.github.com/a47cdd1a013e47b11ab9eb35d4eee0cb">ocamlformat_0_14_1</a></li></li>
    <li><a href="https://gist.github.com/9fdc0937bb5422adab4aeb967fe90259">ocamlformat_0_14_2</a></li></li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/1121ea2d10795edce01d5dc833432e35">ocamlformat</a></li> (ocamlformat_0_16_0)</li>
    <li><a href="https://gist.github.com/8ec3d262a0bfd2c34d2abd4a5572d787">ocamlformat_0_14_3</a></li></li>
    <li><a href="https://gist.github.com/befc3eb4f01a99b73e9b7fdc5721d269">ocamlformat_0_15_0</a></li></li>
    <li><a href="https://gist.github.com/112c114496e2dc9270c47484fc7cd8da">ocamlformat_0_15_1</a></li></li>
  </ul>
</details>

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build ocamlPackages.uucp

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build ocaml-ng.ocamlPackages_4_08.uunf

**Comment by sternenseemann** [comment]
See <https://github.com/dbuenzli/uunf/issues/15> on the aarch64 failure.

File: pkgs/development/ocaml-modules/uucp/default.nix [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
+    ${topkg.buildPhase} --with-cmdliner false --tests ${lib.boolToString doCheck}
+    runHook postBuild
+  '';
 
   inherit (topkg) installPhase;
 
+  inherit doCheck;
+  checkPhase = ''
+    runHook preCheck
+    ocaml -I ${findlib}/lib/ocaml/${ocaml.version}/site-lib pkg/pkg.ml test
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/uucp/default.nix:38]:
```suggestion
    ${topkg.run} test
```

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/uucp/default.nix]: Changed. Thanks, didn't realize `run` wass in `passthru`.

Comment (vbgl) [reply pkgs/development/ocaml-modules/uucp/default.nix]: There are some comments in the source of the `topkg` derivation: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ocaml-modules/topkg/default.nix

File: pkgs/development/ocaml-modules/uucp/default.nix [hunk 2]
```diff
@@ -1,30 +1,45 @@
...
-  version = "11.0.0";
+  version = "13.0.0";
   webpage = "https://erratique.ch/software/${pname}";
+  minimumOCamlVersion = "4.03";
+  doCheck = true;
 in
 
-assert lib.versionAtLeast ocaml.version "4.01";
+assert lib.assertMsg (lib.versionAtLeast ocaml.version minimumOCamlVersion)
+  "${pname} needs at least OCaml ${minimumOCamlVersion}";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/uucp/default.nix:12]: It seems that now `nix-env -qaP` unconditionally outputs:

> trace: uucp needs at least OCaml 4.03

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/uucp/default.nix:12]: Hm, I don't get it with `nix-env -qaP -f ~/src/nix/nixpkgs` and nix 2.3.10.

Comment (vbgl) [reply pkgs/development/ocaml-modules/uucp/default.nix:12]: What about `nix-env -qaP -f ~/src/nix/nixpkgs -A ocaml-ng.ocamlPackages_4_02 > /dev/null`?

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/uucp/default.nix:12]: Completely silent for me and lists uucp.

Comment (vbgl) [reply pkgs/development/ocaml-modules/uucp/default.nix:12]: > Completely silent for me and lists uucp.

This is the behavior before this PR. After, the `uucp` attribute should be gone.

> There's no reason why qaP would evaluate uucp right? it doesn't need meta or similar afaik…

In order to print the names of the derivations, `nix-env` must evaluate their `name` attribute, hence go through the failing assert.

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/uucp/default.nix:12]: Wait, I'm stupid, I was on the wrong branch. Probably not good UX. I wonder why trace and throw messages are treated differently by `nix-env`, but I'll make another PR reverting this change anyways.

**Pull request merged**